bpfd/src/lib.rs

```diff
@@ -48,6 +54,17 @@ pub async fn serve(config: Config, static_programs: Vec<StaticPrograms>) -> anyh
 
     let loader = BpfdLoader::new(tx);
 
+    let mut sigint = signal(SignalKind::interrupt())?;
+    let mut sigterm = signal(SignalKind::terminate())?;
+    let signal_handler = tokio::spawn(async move {
+        select! {
+            _ = sigint.recv() => {debug!("Received SIGINT")},
+            _ = sigterm.recv() => {debug!("Received SIGTERM")},
+        }
+
+        shutdown_tx.send(()).unwrap();
+    });
```

Comment:
```suggestion
    tokio::spawn(async move {
        let mut sigint = signal(SignalKind::interrupt()).unwrap();
        let mut sigterm = signal(SignalKind::terminate()).unwrap();
        loop {
           select! {
                _ = sigint.recv() => {debug!("Received SIGINT")},
                _ = sigterm.recv() => {debug!("Received SIGTERM")},
            }
            shutdown_tx.send(()).unwrap();
        }
    });

```

Reply:
This moves sending the shutdown command based on received signal into its own task.

Reply:
The `loop` part of this change makes it so this task never exits and, in turn, prevents bpfd from stopping as well. Most examples I find in the tokio and tonic docs just let the task finish once the signal is received, alternatively, we would need the main task to be able to remotely stop this with a sync mechanism (but that just seems unnecessarily complicated).